server/db/dbManager.ts

```diff
@@ -24,23 +24,104 @@ export class PostManager {
 		return this.model.findById(postId).exec();
 	};
 
-	async getAllPosts(parent?: mongoose.Types.ObjectId, depth: number = 0, parentObject = { children: [] }): Promise<models.IPost[]> {
-		let posts;
-		if (parent) {
-			posts = await this.model.find({ parent }).sort({ upvotes: -1 }).lean().exec();
-			if (depth < parseInt(process.env.MAX_COMMENT_DEPTH)) {
-				for (const post of posts) {
-					post.children = [];
-					if (depth > 0) {
-						parentObject.children.push(post);
+	async getAllPosts(recursive: boolean, parent?: mongoose.Types.ObjectId, parentObject = { children: [] }): Promise<object> {
+		if (parent && recursive) {
+			const count = await this.model.find({ parent }).countDocuments();
+			if (count === 0) {
+				return [];
+			}
+			const parentId = new ObjectId(parent);
+			const post = await this.model.aggregate([
+				{
+					$graphLookup: {
+						from: 'posts',
+						startWith: '$_id',
+						connectFromField: '_id',
+						connectToField: 'parent',
+						as: 'children',
+						maxDepth: parseInt(process.env.MAX_COMMENT_DEPTH) - 1,
+						depthField: 'level'
+					}
+				},
```

Comment:
You definitely need to stick with the same library. So if you use mongoose, so you need to make this the same query using mongoose.